activerecord/CHANGELOG.md

```diff
@@ -1,3 +1,17 @@
+*   Introduce ApplicationRecord, an Active Record layer supertype
+
+    It's pretty common for folks to monkey patch `ActiveRecord::Base` to
+    work around an issue or introduce extra functionality. Instead of
+    shoving even more stuff in `ActiveRecord::Base`, `ApplicationRecord` can
+    hold all those custom work the apps may need.
```

Comment:
My point wasn't whether or not we should advertise to engine authors. It was that the motivation stated here isn't why we want to add this class.